### PR TITLE
🐛 fix: changes to release process

### PR DESCRIPTION
**Pull request by richardcase**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Changes from the failed releases:

- Including the `toolchain` directive
- Bumping the GCB image so that it has a newer Go version (that understands the `toolchain` directive)
- Changing the clusterawsadm build to be inline with CAPI

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Relates #4812 

**Special notes for your reviewer**:

This will be done for realease-2,3 manually instead of via automatic cherry pick

**Checklist**:

- x ] squashed commits
- [ ] includes documentation
- [x] includes [emojis](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Various changes to the releases process.
```

### CODE REVIEW

**Review by damdo — COMMENTED** [review]
/lgtm
Thanks!

File: Makefile [hunk 1]
```diff
@@ -607,18 +607,17 @@ release-binaries: ## Builds the binaries to publish with a release
...
+release-binary: $(RELEASE_DIR) versions.mk ## Release binary
 	docker run \
 		--rm \
 		-e CGO_ENABLED=0 \
 		-e GOOS=$(GOOS) \
 		-e GOARCH=$(GOARCH) \
-		--mount=source=gocache,target=/go/pkg/mod \
-		--mount=source=gocache,target=/root/.cache/go-build \
+		-e GOCACHE=/tmp/ \
+		--user $$(id -u):$$(id -g) \
```
Comment (faiq) [inline Makefile:617]: Why do we need this?

Comment (richardcase) [reply Makefile:617]: This is to be consistent with upstream CAPI (i.e. when it builds clusterctl) and it also us having to use the git config command.

**Pull request merged**